### PR TITLE
Replace table column Link useLabel with label

### PR DESCRIPTION
**Pull request by georgehristov**
allow to set custom label template

### CODE REVIEW

File: src/Table/Column/Link.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ class Link extends Table\Column
      */
     public $args = [];
 
-    /** @var bool use value as label of the link */
+    /** @var bool|string use value as label of the link */
     public $use_label = true;
```
Comment (mvorisek) [inline src/Table/Column/Link.php]: no string for "use_xxx"

Comment (georgehristov) [reply src/Table/Column/Link.php]: I would normally propose using $label.
`$label = false` - no label
`$label = true` - default label template
`$label = 'string'` - use it as template

_(It works pretty nice in the new filestore Upload Control having such feature)_

Comment (mvorisek) [reply src/Table/Column/Link.php]: yes

File: src/Table/Column/Link.php [hunk 2]
```diff
@@ -136,7 +136,11 @@ public function getDataCellTemplate(\atk4\data\Field $field = null)
 
         $label = '';
         if ($this->use_label) {
-            $label = $field ? ('{$' . $field->short_name . '}') : '[Link]';
+            if ($this->use_label === true) {
+                $label = $field ? ('{$' . $field->short_name . '}') : '[Link]';
```
Comment (mvorisek) [inline src/Table/Column/Link.php]: `{$` is there for some templating?

Comment (georgehristov) [reply src/Table/Column/Link.php]: yes, $label is a template string

File: src/Table/Column/Link.php [hunk 3]
```diff
@@ -55,9 +55,20 @@ class Link extends Table\Column
...
     public $args = [];
 
-    /** @var bool|string use value as label of the link */
+    /** @deprecated use $label instead - will be removed dec-2020 */
     public $use_label = true;
 
+    /**
+     * Label template to use on the link.
+     *
+     * $label = true - use field value as label
```
Comment (mvorisek) [inline src/Table/Column/Link.php:64]: implied from type!

File: src/Table/Column/Link.php (hunk 2 above)
Comment (mvorisek) [reply src/Table/Column/Link.php]: then it is set wrong - label can be be used as template...

Comment (georgehristov) [reply src/Table/Column/Link.php]: I didn't get your comment?

Comment (mvorisek) [reply src/Table/Column/Link.php]: imagine label with `{$xx}`

Comment (georgehristov) [reply src/Table/Column/Link.php]: This is true for any decorator, then you need to escape the template like syntax

Comment (mvorisek) [reply src/Table/Column/Link.php]: and html? `<script>alert("hi, my Bitcoin wallet is...");</script>`

Comment (georgehristov) [reply src/Table/Column/Link.php]: scripts gets directly executed on each row render

Comment (mvorisek) [reply src/Table/Column/Link.php]: I really hate these single purpose solution and even more if they are not complete/secure at all.

image

can we solve it better? is this for some core atk4 feature?

Comment (georgehristov) [reply src/Table/Column/Link.php]: The additional code is due to BC notices, etc. Actually additional code is 1 line.
It provides much more flexible functionality for the Link where dev can set whatever template he likes, not just the field value

Comment (mvorisek) [reply src/Table/Column/Link.php]: but implemented as single purpose if - should we copy paste that 100x times?!?

**Comment by mvorisek** [comment]
In `Card::useLabel` and `CardDeck::useLabel` are simillar properties. I am still not fully sold on this, especially as the (newly introduced) `Link::label` requires a template, which requires a `$fieldShortName` - do we need/want to allow to link across other field values?

@georgehristov please provide some usecases

**Comment by mvorisek** [comment]
Currently this PR is missing a test, not encoding HTML in the label and it is a questionable if this is even wanted, as the label currently represents the value itself.

As no feedback/usecases were provided, I have to close this PR.

**Pull request closed without merging**